File: packages/php/src/Types/Metadata/StructuredData.php
<?php

declare(strict_types=1);

namespace Kreuzberg\Types\Metadata;

class StructuredData
{
    public string $dataType;
    public string $rawJson;
    public ?string $schemaType = null;

    public function __construct(
        string $dataType,
        string $rawJson,
        ?string $schemaType = null,
    ) {
        $this->dataType = $dataType;
        $this->rawJson = $rawJson;
        $this->schemaType = $schemaType;
    }
}

File: packages/java/src/main/java/dev/kreuzberg/types/metadata/StructuredData.java
package dev.kreuzberg.types.metadata;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.jspecify.annotations.Nullable;

/**
 * Structured data (JSON-LD, microdata, RDFa) metadata.
 */
public record StructuredData(
    @JsonProperty("data_type") String dataType,
    @JsonProperty("raw_json") String rawJson,
    @JsonProperty("schema_type") @Nullable String schemaType
) { }
